src/compiler/compiler.rs

```diff
@@ -1173,10 +1188,7 @@ mod test {
         let f = TestFixture::new();
         let creator = new_creator();
         let pool = ThreadPool::sized(1);
-        next_command(
-            &creator,
-            Ok(MockChild::new(exit_status(0), "clang\nfoo", "")),
-        );
+        next_command(&creator, Ok(MockChild::new(exit_status(0), "clang\n", "")));
         let c = detect_compiler(creator, &f.bins[0], f.tempdir.path(), &[], &pool, None)
```

Comment:
Should we modify this test to check for `version`?

Reply:
I guess I could add a test that checks that cache keys for 2 different versions are different...